### PR TITLE
ACMS-1272: Enable modules in a batch when enabling starter kits via Cloud UI and Site Factory UI

### PR DESCRIPTION
**Pull request by rajeshreeputra**
**Motivation**
Fixes #ACMS-1272

**Proposed changes**
Enable modules in batch when enabling via UI using starterkit.

**Alternatives considered**
NA

**Testing steps**
- Use this branch.
- Install site `drush si minimal`
- Install tour module `drush in acquia_cms_tour`
- Login into application and head towards `/admin/tour/dashboard`
- Click on `Starter kit set-up`
- Select starter kit click on next
- Select Demo content or content model click on complete
- It should start batch and install modules
- Verify site working fine, all modules installed successfully.

**Merge requirements**
- [ ] _Major change_, _Minor change_, _Bug_, _Enhancement_, and/or _Chore_ label applied
- [ ] Manual testing by a reviewer

### CODE REVIEW

File: modules/acquia_cms_tour/src/Plugin/AcquiaCmsStarterKit/StarterKitConfigForm.php [hunk 1]
```diff
@@ -204,7 +209,7 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
       $this->state->set('show_starter_kit_modal', FALSE);
       $this->state->set('starter_kit_wizard_completed', TRUE);
       $service->enableModules($starter_kit, $starter_kit_demo, $starter_kit_content_model);
-      $this->messenger()->addStatus('The required starter kit has been installed. Also, the related modules & themes have been enabled.');
+      $this->messenger()->addStatus('The ' . $starter_kits[$starter_kit] . ' starter kit has been installed. Also, the related modules & themes have been enabled.');
```
Comment (apathak18) [inline modules/acquia_cms_tour/src/Plugin/AcquiaCmsStarterKit/StarterKitConfigForm.php:212]: @rajeshreeputra use $this->messenger()->addStatus($this->t('The  %starter_kit starter kit has been installed. Also, the related modules & themes have been enabled.', ['%starter_kit' => $starter_kits[$starter_kit]]))

File: modules/acquia_cms_tour/src/Services/StarterKitService.php [hunk 2]
```diff
@@ -91,6 +122,16 @@ public function enableModules(string $starter_kit, string $demo_question = NULL,
...
   }
 
+  /**
+   * Handler for enabling modules.
+   *
+   * @param string $module
+   *   Variable holding the module name.
+   */
+  public static function enableModule(string $module) {
+    \Drupal::service('module_installer')->install([$module]);
```
Comment (apathak18) [inline modules/acquia_cms_tour/src/Services/StarterKitService.php:132]: @rajeshreeputra  As this file is already a service then please Use D.I (use Drupal\Core\Extension\ModuleInstallerInterface;)

File: modules/acquia_cms_tour/src/Services/StarterKitService.php [hunk 3]
```diff
@@ -91,6 +122,16 @@ public function enableModules(string $starter_kit, string $demo_question = NULL,
       ->save();
   }
 
+  /**
+   * Handler for enabling modules.
+   *
+   * @param string $module
+   *   Variable holding the module name.
+   */
+  public static function enableModule(string $module) {
```
Comment (apathak18) [inline modules/acquia_cms_tour/src/Services/StarterKitService.php:131]: enableModules and enableModule isn't confusing at first go?
Can we rename this functions **enableSingleModule** or something different? @rajeshreeputra

File: modules/acquia_cms_tour/src/Services/StarterKitService.php (hunk 2 above)
Comment (rajeshreeputra) [reply modules/acquia_cms_tour/src/Services/StarterKitService.php:132]: in static function DI isn't allowed

File: modules/acquia_cms_tour/src/Services/StarterKitService.php (hunk 3 above)
Comment (rajeshreeputra) [reply modules/acquia_cms_tour/src/Services/StarterKitService.php]: done

File: modules/acquia_cms_tour/src/Plugin/AcquiaCmsStarterKit/StarterKitConfigForm.php (hunk 1 above)
Comment (rajeshreeputra) [reply modules/acquia_cms_tour/src/Plugin/AcquiaCmsStarterKit/StarterKitConfigForm.php]: done

File: modules/acquia_cms_tour/src/Plugin/AcquiaCmsStarterKit/StarterKitConfigForm.php [hunk 4]
```diff
@@ -189,25 +189,34 @@ public function buildForm(array $form, FormStateInterface $form_state) {
...
     $starter_kit_content_model = $form_state->getValue(['content_model']) ?? 'No';
     if ($starter_kit_demo && $starter_kit_content_model) {
       $this->state->set('acquia_cms.starter_kit_demo', $starter_kit_demo);
       $this->state->set('acquia_cms.starter_kit_content_model', $starter_kit_content_model);
       $this->state->set('acquia_cms_tour_staretr_kit_demo_progress', TRUE);
-      $this->messenger()->addStatus('The configuration options have been saved.');
+      $this->messenger()->addStatus($this->t('The configuration options have been saved.'));
     }
     $starter_kit = $this->state->get('acquia_cms.starter_kit');
     $service = \Drupal::service('acquia_cms_tour.starter_kit');
```
Comment (apathak18) [inline modules/acquia_cms_tour/src/Plugin/AcquiaCmsStarterKit/StarterKitConfigForm.php:206]: use D.I here

**Review by vishalkhode1 — APPROVED** [review]

**Pull request merged**